sdk/rust/src/http.rs

```diff
@@ -482,46 +510,119 @@ impl ResponseOutparam {
         self,
         response: OutgoingResponse,
         buffer: Vec<u8>,
-    ) -> anyhow::Result<()> {
+    ) -> Result<(), Error> {
         use futures::SinkExt;
         let mut body = response.take_body();
         self.set(response);
         body.send(buffer).await
     }
 }
 
-/// Send an outgoing request
-pub async fn send<I, O>(request: I) -> Result<O, SendError>
+/// Send a request and get a response
+pub async fn send<S, R>(request: S) -> Result<Result<R, R::Error>, Error>
```

Comment:
What made this necessary? Are there conversions that can't do the boxed errors?

Reply:
No, it was a (admittedly fairly unsuccessful) attempt at simplifying the signature.

Reply:
Yeah doesn't seem worth the doubled Results.